### PR TITLE
Popup in compact interactive list

### PR DESCRIPTION
**Pull request by adoroshk**
### Summary
Terra compact interactive list has to support terra-popup in its cell. As compact interactive list has its own key and focus event handlers, terra-popup content key and focus events propagation needs to be stopped in order for compact interactive list to preserve key navigation between cells.

This PR:
1. Stops key and focus event propagation in terra-popup content.
2. Updates terra-compact-interactive-list documentation and cell content example with popup inside of a cell.
3. Updates terra-compact-interactive-list test example, adds new wdio tests and updates old ones.
4. Updates jest snapshots for terra-popup.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

**This PR resolves:**

UXPLATFORM-10389

### CODE REVIEW

File: packages/terra-popup/src/_PopupContent.jsx [hunk 1]
```diff
@@ -237,6 +237,8 @@ class PopupContent extends React.Component {
             onResize={this.handleOnResize}
             refCallback={refCallback}
             role={popupContentRole || null}
+            onKeyDown={event => event.stopPropagation()} // Added for Popup support in terra-compact-interactive-list. As focus trap doesn't stop key press event propagation to the CIL cell, it interferes with cell key press event handler.
```
Comment (cm9361) [inline packages/terra-popup/src/_PopupContent.jsx:240]: You should create a function will a useCallback hook for memoization to prevent unnecessary rerenders.  Same comment for your new onFocus function.

File: packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx [hunk 2]
```diff
@@ -34,7 +36,56 @@ const FeaturedIcon = () => {
...
+  const setButtonNode = (node) => { buttonRef.current = node; };
+  const getButtonNode = () => buttonRef.current;
+  const handleButtonClick = () => { setIsOpen(true); };
+  const handleRequestClose = () => { setIsOpen(false); };
+  const closeAndSetFocus = () => {
+    handleRequestClose();
+    buttonRef.current.focus();
+  };
+
+  const handleOnChange = (e) => {
```
Comment (cm9361) [inline packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx:59]: This should have a useCallback hook

File: packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx [hunk 3]
```diff
@@ -34,7 +36,56 @@ const FeaturedIcon = () => {
...
+
+const PopupWithInteractiveContent = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selected, setSelected] = useState(null);
+  const buttonRef = useRef(null);
+
+  const setButtonNode = (node) => { buttonRef.current = node; };
+  const getButtonNode = () => buttonRef.current;
+  const handleButtonClick = () => { setIsOpen(true); };
+  const handleRequestClose = () => { setIsOpen(false); };
```
Comment (cm9361) [inline packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx:53]: These should have a useCallback hook.

File: packages/terra-popup/src/_PopupContent.jsx (hunk 1 above)
Comment (adoroshk) [reply packages/terra-popup/src/_PopupContent.jsx]: Fixed here: https://github.com/cerner/terra-framework/pull/2149/commits/0dc66ba494b2880e5c257785e1d85fabc43c1632

[the remark this answers is not in the document]
Comment (adoroshk) [reply packages/terra-popup/src/_PopupContent.jsx]: The keys that might mess with the CIL are Tab and Arrow keys (Right/Left/Up/Down). However, as the key event propagation has been stopped on the popup content wrapper `<div>`, it only stops bubbling up to the CIL. The key event for all keys (including TAB) will still work inside the popup the same way they did before.

File: packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx (hunk 2 above)
Comment (adoroshk) [reply packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx]: Callback added: https://github.com/cerner/terra-framework/pull/2149/commits/cacc605c3c50d58642b90fe8cef6d04dec530c9c

File: packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx (hunk 3 above)
Comment (adoroshk) [reply packages/terra-framework-docs/src/terra-dev-site/test/compact-interactive-list/CellContent.test.jsx]: Callback added: https://github.com/cerner/terra-framework/pull/2149/commits/cacc605c3c50d58642b90fe8cef6d04dec530c9c

File: packages/terra-popup/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Added event handlers to call `stopPropagation()` on key and focus events on PopupContent for `terra-popup` support in `terra-compact-interactive-list` cell.
```
Comment (sdadn) [inline packages/terra-popup/CHANGELOG.md:6]: can this feature be used outside of compact interactive list?

Comment (adoroshk) [reply packages/terra-popup/CHANGELOG.md:6]: It works everywhere, doesn't matter inside of CIL or outside of it, as outside of the compact interactive list it makes no difference. I just wanted to provide a reason why it was added, as it makes no difference for all the cases when popoup is outside of CIL (or any other component that has key navigation).

Comment (sdadn) [reply packages/terra-popup/CHANGELOG.md:6]: Then I would reword this to be something generic like:

```suggestion
  * Added event handlers to call `stopPropagation()` on key and focus events on PopupContent in other components.
```

**Review by sdadn — APPROVED** [review]
